src/FixedDataTableBufferedRows.js

```diff
@@ -78,36 +79,30 @@ class FixedDataTableBufferedRows extends React.Component {
     var rowClassNameGetter = props.rowClassNameGetter || emptyFunction;
     var rowsToRender = this.props.rowsToRender || [];
 
-    this._staticRowArray.length = rowsToRender.length;
-    var baseOffsetTop = props.offsetTop - props.scrollTop;
+    if (!props.isScrolling) {
+      this._staticRowArray.length = this.props.rowsToRender.length;
+    } else {
+      // we are scrolling, so don't display any rows which lie outside the viewport
+      this._staticRowArray.forEach((row, i) => {
+        const rowOutsideViewport = !inRange(row.props.index, this.props.firstRowIndex, this.props.endRowIndex);
+        if (rowOutsideViewport) {
+          this._staticRowArray[i] = React.cloneElement(this._staticRowArray[i], {
+            visible: false,
+          });
+        }
+      });
+    }
 
-    for (let i = 0; i < rowsToRender.length; ++i) {
-      const rowIndex = rowsToRender[i];
-      if (rowIndex === undefined) {
```

Comment:
Since we use a forEach, only defined values (rows) are iterated over.

Earlier, since we used a for loop, we got undefined values. To get over this (just my guess), we render a "fake row" by rendering the row with the visible attribute set to false. This will let the row (at the implementation level) be null. But this should still save an unmount/mount for the row wrapper.

So, I'll add it and have a function return the fake row, which seems cleaner IMO.